v2/sourcedb-to-spanner/src/test/java/com/google/cloud/teleport/v2/templates/PipelineControllerTest.java

```diff
@@ -136,29 +145,132 @@ public void setup() {
             .end()
             .endTable()
             .build();
+
+    // Create a dummy schema overrides file for tests that need it
+    schemaOverridesFile = temporaryFolder.newFile("schema_overrides.json").toPath();
+    String overridesJsonContent =
+        "{\n"
+            + "  \"renamedTables\": {\n"
+            + "    \"source_table\": \"spanner_table\"\n"
+            + "  }\n"
+            + "}";
+    try (BufferedWriter writer = Files.newBufferedWriter(schemaOverridesFile)) {
+      writer.write(overridesJsonContent);
```

Comment:
Please consider calling [flush](https://docs.oracle.com/javase/8/docs/api/java/io/BufferedWriter.html#flush--) after the write,
W/o flush, many times the file remains cached in memory and does not get synced to disk.
In one of the C* Its I had noticed flakiness before submitting, which needed flush.